Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/repository/storage/ContainerPostProcessor.kt

```diff
@@ -0,0 +1,47 @@
+package de.rki.coronawarnapp.vaccination.core.repository.storage
+
+import com.google.gson.Gson
+import com.google.gson.TypeAdapter
+import com.google.gson.TypeAdapterFactory
+import com.google.gson.reflect.TypeToken
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonWriter
+import dagger.Reusable
+import de.rki.coronawarnapp.vaccination.core.qrcode.VaccinationCertificateCOSEParser
+import de.rki.coronawarnapp.vaccination.core.server.proof.ProofCertificateCOSEParser
+import timber.log.Timber
+import java.io.IOException
+import javax.inject.Inject
+
+@Reusable
+class ContainerPostProcessor @Inject constructor(
+    private val vaccinationCertificateCOSEParser: VaccinationCertificateCOSEParser,
+    private val proofCertificateCOSEParser: ProofCertificateCOSEParser,
+) : TypeAdapterFactory {
```

Comment:
Allows us to add extra data to classes right after deserialization. We don't have to hardcode the parsers anymore.